packages/swing-store/src/swingStore.js

```diff
@@ -82,6 +99,170 @@ export function makeSnapStoreIO() {
+ * exporter may be used while another SwingStore instance is active for the same
+ * DB, possibly in another thread or process.  It guarantees that regardless of
+ * the concurrent activity of other swingStore instances, the data representing
+ * the commit point will stay consistent and available.
+ *
+ * @property {() => Iterable<KVPair>} getKVData
+ *
+ * Get a full dump of KV data from the swingStore. This represents both the
+ * KVStore (excluding host and local prefixes), as well as any data needed to
+ * validate all artifacts, both current and historical. As such it represents
+ * the root of trust for the application.
+ *
+ * Content of validation data (with supporting entries for indexing):
+ * - export.snapshot.${vatID}.${endPos} = ${{ vatID, endPos, hash, size });
+ * - export.snapshot.${vatID}.current = `snapshot.${vatID}.${endPos}`
+ * - export.transcript.${vatID}.${startPos} = ${{ vatID, startPos, endPos, hash, size }}
+ * - export.transcript.${vatID}.current = ${{ vatID, startPos, endPos, hash, size }}
+ *
+ * @property {(includeHistorical: boolean) => Iterable<string>} getArtifactNames
+ *
+ * Get a list of name of artifacts available from the swingStore.  A name returned
+ * by this method guarantees that a call to `getArtifact` on the same exporter
+ * instance will succeed. Options control the filtering of the artifact names
+ * yielded.
+ *
+ * Artifact names:
+ * - transcript.${vatID}.${startPos}.${endPos}
+ * - snapshot.${vatID}.${endPos}
+ *
+ * @property {(name: string) => AsyncIterable<Uint8Array>} getArtifact
+ *
+ * Retrieve an artifact by name.  May throw if the artifact is not available,
+ * which can occur if the artifact is historical and wasn't been preserved.
+ *
+ * @property {() => Promise<void>} close
+ *
+ * Dispose of all resources held by this exporter. Any further operation on this
+ * exporter or its outstanding iterators will fail.
+ */
+
+/**
+ * @param {string} dirPath
+ * @returns {SwingStoreExporter}
+ */
+export function makeSwingStoreExporter(dirPath) {
+  assert.typeof(dirPath, 'string');
+  const filePath = path.join(dirPath, 'swingstore.sqlite');
+  const db = sqlite3(filePath);
+
+  // Execute the data export in a (read) transaction, to ensure that we are
+  // capturing the state of the database at a single point in time.
+  const sqlBeginTransaction = db.prepare('BEGIN IMMEDIATE TRANSACTION');
+  sqlBeginTransaction.run();
+
+  const snapStore = makeSnapStore(db, makeSnapStoreIO());
+  const transcriptStore = makeTranscriptStore(
+    db,
+    () => {},
+    () => {},
+  );
+
+  const sqlGetAllKVData = db.prepare(`
+    SELECT key, value
+    FROM kvStore
+    ORDER BY key
+  `);
+
+  /**
+   * @returns {Iterable<KVPair>}
+   * @yields {KVPair}
+   */
+  function* getKVData() {
+    const kvPairs = sqlGetAllKVData.iterate();
+    for (const kv of kvPairs) {
+      if (getKeyType(kv.key) === 'consensus') {
+        yield [kv.key, kv.value];
```

Comment:
Why not reserve `export.` in a similar way `host.` is reserved?

Reply:
In discussion, @warner and I concluded that having the keyspace be shaped like:
```
kv.${kvstorekey}
transcript.${transcriptything}
snapshot.${snapshottything}
```
would be better than
```
${whateveraslongasitdoesntstartwithexportdot}
export.transcript.${transcriptything}
export.snapshot.${snapshottything}
```
because (1) it's a little safer from hypothetical attacks where somebody is somehow able to provide an arbitrary kvStore key, (2) by allowing each subtree of the namespace to be handled separately, we're a little more future proof as we add possible new artifact types (e.g., bundles) and possible new kv-ish things (e.g., c-list entries that go into their own table).

While `.host` and `.local` _are_ special reserved kvStore key prefixes, they still denote things that are actually put into the kvStore.

Anyway, this is now done.